core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/component/ComponentDtoRepository.java

```diff
@@ -0,0 +1,179 @@
+                "INNER JOIN " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ComponentVersions").getName() + " versions " +
+                "on component_designs.COMP_ID = versions.COMP_ID" +
+                getWhereClause(componentFilters) +
+                getOrderByClause(pageable) +
+                getLimitAndOffsetClause(pageable);
+    }
+
+    @Override
+    public Page<ComponentDto> getAll(Authentication authentication, Pageable pageable, List<ComponentFilter> componentFilters) {
+        try {
+            Map<ComponentKey, ComponentDtoBuilder> componentDtoBuilders = new LinkedHashMap<>();
+            String query = getFetchAllQuery(authentication, pageable, componentFilters);
+            CachedRowSet cachedRowSet = metadataRepositoryConfiguration.getDesignMetadataRepository().executeQuery(query, "reader");
+
+            while (cachedRowSet.next()) {
+                mapRow(cachedRowSet, componentDtoBuilders);
+            }
+            List<ComponentDto> componentDtoList = componentDtoBuilders.values().stream()
+                    .map(ComponentDtoBuilder::build)
+                    .collect(Collectors.toList());
+            return new PageImpl<>(componentDtoList, pageable, getRowSize(authentication));
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private long getRowSize(Authentication authentication) throws SQLException {
+        String query = "select count(*) as row_count from (select distinct component_designs.COMP_ID, versions.COMP_VRS_NB " +
+                "from " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("Components").getName() + " component_designs " +
+                "INNER JOIN " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ComponentVersions").getName() + " versions " +
+                "on component_designs.COMP_ID = versions.COMP_ID );";
+        CachedRowSet cachedRowSet = metadataRepositoryConfiguration.getDesignMetadataRepository().executeQuery(query, "reader");
+        cachedRowSet.next();
+        return cachedRowSet.getLong("row_count");
+    }
+
+    private String getOrderByClause(Pageable pageable) {
+        if (pageable.getSort().isUnsorted()) return " ";
+        List<String> sorting = pageable.getSort().stream().map(order -> {
+            if (order.getProperty().equalsIgnoreCase("NAME")) {
+                return "component_designs.COMP_NM" + " " + order.getDirection();
+            } else if (order.getProperty().equalsIgnoreCase("VERSION")) {
+                return "versions.COMP_VRS_NB" + " " + order.getDirection();
+            } else {
+                return null;
+            }
+        })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        if (sorting.isEmpty()) {
+            return "";
+        }
```

Comment:
Shouldn't you add a default sorting parameter? E.g. COMP_ID

Reply:
No, if there are no sorting, the request is empty and we do nothing.
The method return the query to sort like : Order by ...

If a sorting exists : "select * from ENTITY order by NAME ASC" because it returns the "order by" clause
If there is nothing to sort : "select * from ENTITY" because it returns empty string

Reply:
okay, but I think some database technologies don't support consistent behavior regarding order when no 'order by' clause is provided. As a consequence, second page api call might not start where first page api call left off.  

https://stackoverflow.com/questions/61068381/what-is-the-default-sort-of-paging-and-sorting-repository-in-spring-jpa
https://dba.stackexchange.com/questions/6051/what-is-the-default-order-of-records-for-a-select-statement-in-mysql

Reply:
Okay, I will update it.

Should I also to add default sorting for the script repository ? Because it also returns empty string